src/Diagnostics.RuntimeHost/Services/HealthCheckService/HealthCheckService.cs

```diff
@@ -61,24 +69,61 @@ public async Task RunHealthCheck()
             await RetryHelper.RetryAsync(HealthCheckPing, "Healthping", "", 3, 100);
         }
 
+        [System.Diagnostics.CodeAnalysis.SuppressMessage("Reliability", "CA2007:Consider calling ConfigureAwait on the awaited task", Justification = "<Pending>")]
         public async Task<IEnumerable<HealthCheckResult>> RunDependencyCheck(DataProviders.DataProviders dataProviders)
         {
-            var healthCheckResultsTask = new List<Task<HealthCheckResult>>();
+            if (cache.TryGetValue(DEPENDENCYCHECK_CACHE_KEY, out object cachedDependencyChecks))
+            {
+                return (IEnumerable<HealthCheckResult>)cachedDependencyChecks;
+            }
+
```

Comment:
nitpick : After the if block, How about adding a log here to know how frequent is the cache miss and the dependency call is actually running?